### PR TITLE
fix: exited container fetch metrics cause error

### PR DESCRIPTION
**Pull request by gsanchezgavier**
Exited container metrics fetch was causing the follow error:
```log
error fetching metrics for container <container_id>: failed to parse cgroups2 paths, error: failed to open file: /proc/0/cgroup, while detecting cgroup paths error: open /proc/0/cgroup: no such file or directory
```
Since metrics still were tried to be collected from an exited container where Pid changes from the original value to `0`, which does not represent a real Pid but a condition of that container (exited state).

Changes:
- Metrics fetch is no longer executed for exited containers, expired or not.
- Parsing error related to time since exited are only returned in case `TTL != 0`

Note: 
The process function is a bit confusing since return values doesn't follow a clear standar of (value,error) . I didn't want to perform a bigger refactor modifying that behaviour for just fixing a error log.

### CODE REVIEW

**Review by sigilioso — APPROVED** [review]
Thanks for taking care of this, I've left a comment

File: src/nri/sampler.go [hunk 1]
```diff
@@ -92,11 +93,15 @@ func (cs *ContainerSampler) SampleAll(ctx context.Context, i *integration.Integr
...
+			if errors.Is(err, biz.ErrExitedContainerExpired) {
+				log.Debug("skipping samples for container (%s): %s", container.ID, err.Error())
 				continue
 			}
-			log.Error("error fetching metrics for container %v: %v", container.ID, err)
+			if errors.Is(err, biz.ErrExitedContainerUnexpired) {
+				log.Debug("skipping fetching metrics from container (%s): %s", container.ID, err.Error())
+			}
+
+			log.Error("fetching metrics for container %v: %v", container.ID, err)
```
[the remark this answers is not in the document]
Comment (gsanchezgavier) [reply src/nri/sampler.go:104]: 🤦  thanks

[the remark this answers is not in the document]
Comment (gsanchezgavier) [reply src/nri/sampler.go]: done

**Review by sigilioso — APPROVED** [review]
LGTM 👍

**Pull request merged**